src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -0,0 +1,181 @@
+#---------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#---------------------------------------------------------------------------------------------
+
+import os
+from six.moves import configparser
+
+from azure.cli.core._config import GLOBAL_CONFIG_DIR
+
+CUSTOM_CLOUD_CONFIG_FILE = os.path.join(GLOBAL_CONFIG_DIR, 'clouds.config')
+
+class CloudNotRegisteredException(Exception):
+    def __init__(self, cloud_name):
+        super(CloudNotRegisteredException, self).__init__(cloud_name)
+        self.cloud_name = cloud_name
+    def __str__(self):
+        return "The cloud '{}' is not registered.".format(self.cloud_name)
+
+class CloudAlreadyRegisteredException(Exception):
+    def __init__(self, cloud_name):
+        super(CloudAlreadyRegisteredException, self).__init__(cloud_name)
+        self.cloud_name = cloud_name
+    def __str__(self):
+        return "The cloud '{}' is already registered.".format(self.cloud_name)
+
+class CannotUnregisterCloudException(Exception):
+    pass
+
+class CloudEndpointUrl(object): # pylint: disable=too-few-public-methods
+    MANAGEMENT = 'management'
+    RESOURCE_MANAGER = 'resource_manager'
+    SQL_MANAGEMENT = 'sql_management'
+    GALLERY = 'gallery'
+    ACTIVE_DIRECTORY = 'active_directory'
+    ACTIVE_DIRECTORY_RESOURCE_ID = 'active_directory_resource_id'
+    ACTIVE_DIRECTORY_GRAPH_RESOURCE_ID = 'active_directory_graph_resource_id'
+
+class CloudParam(object): # pylint: disable=too-few-public-methods
```

Comment:
Same general comments as above regarding `__init__` and lower casing the property names. Also I would recommend naming the class CloudSuffixes since they are all URL suffixes and Param is pretty vague.

Also, the default strings here and above could just as well be empty strings because 'active_directory' isn't a valid URL suffix for anything.